packages/server-renderer/src/renderToString.ts

```diff
@@ -176,7 +188,37 @@ function renderVNode(
       push(`<!---->`)
       break
     case Portal:
-      // TODO
+      if (!vnode.props || !vnode.props.target) {
+        console.warn(`[@vue/server-renderer] Portal doesn't have target prop`)
+        break
+      }
+      if (!isString(vnode.props.target)) {
+        console.warn(
+          `[@vue/server-renderer] Portal target must be a query selector`
+        )
+        break
+      }
+
+      push(
+        (async () => {
+          const { buffer: content, push: portalPush } = createBuffer()
+
+          portalPush(`<!---->`)
```

Comment:
Portal content is always rendered as direct children of the target element so there's no need to wrap with comment nodes.